src/vis/vis.js

```diff
@@ -262,29 +268,46 @@ var VisModel = Backbone.Model.extend({
     }
   },
 
+  init: function (options) {
```

Comment:
(I'm not sure if the `options.success` callback that we're passing to `instantiateMap` would need to be called after the first or the second map reload when there're active filters.).